CMakeLists.txt

```diff
@@ -0,0 +1,459 @@
+                      CUDA_STANDARD                       17
+                      CUDA_STANDARD_REQUIRED              ON
+                      LIBRARY_OUTPUT_DIRECTORY            lib)
+
+target_link_libraries(cunumeric
+   PUBLIC legate::core
+          $<TARGET_NAME_IF_EXISTS:NCCL::NCCL>
+  PRIVATE BLAS::BLAS
+          tblis::tblis
+          # Add Conda library and include paths
+          $<TARGET_NAME_IF_EXISTS:conda_env>
+          $<TARGET_NAME_IF_EXISTS:OpenMP::OpenMP_CXX>
+          $<TARGET_NAME_IF_EXISTS:cutensor::cutensor>)
+
+# Change THRUST_DEVICE_SYSTEM for `.cpp` files
+if(Legion_USE_OpenMP)
+  list(APPEND CUNUMERIC_CXX_OPTIONS -UTHRUST_DEVICE_SYSTEM)
+  list(APPEND CUNUMERIC_CXX_OPTIONS -DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_OMP)
+elseif(NOT Legion_USE_CUDA)
+  list(APPEND CUNUMERIC_CXX_OPTIONS -UTHRUST_DEVICE_SYSTEM)
+  list(APPEND CUNUMERIC_CXX_OPTIONS -DTHRUST_DEVICE_SYSTEM=THRUST_DEVICE_SYSTEM_CPP)
+endif()
+
+###
+# Preinclude `realm_defines.h` and `legion_defines.h` in all compilations.
+# TODO: It isn't possible to expose this as part of the public legate_core
+# target's interface in a portable way, so this code needs to be duplicated
+# in both legate.core and cuNumeric.
+#
+# A better solution would be to add #include directives to the requisite
+# legate.core and cuNumeric headers.
+###
+include(cmake/Modules/legion_helpers.cmake)
+get_legion_and_realm_includes(extra_include_options)
+
+target_compile_options(cunumeric
+  PRIVATE ${extra_include_options}
+          "$<$<COMPILE_LANGUAGE:CXX>:${CUNUMERIC_CXX_OPTIONS}>"
+          "$<$<COMPILE_LANGUAGE:CUDA>:${CUNUMERIC_CUDA_OPTIONS}>")
+
+target_compile_definitions(cunumeric
+  PUBLIC  "$<$<COMPILE_LANGUAGE:CXX>:${CUNUMERIC_CXX_DEFS}>"
+          "$<$<COMPILE_LANGUAGE:CUDA>:${CUNUMERIC_CUDA_DEFS}>")
+
+target_include_directories(cunumeric
+  PRIVATE
+    $<BUILD_INTERFACE:${CUNUMERIC_SOURCE_DIR}/src>
+  INTERFACE
+    $<INSTALL_INTERFACE:include>
+)
+
+if(Legion_USE_CUDA)
+  file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/fatbin.ld"
+[=[
+SECTIONS
+{
+.nvFatBinSegment : { *(.nvFatBinSegment) }
+.nv_fatbin : { *(.nv_fatbin) }
+}
+]=])
+
+  # ensure CUDA symbols aren't relocated to the middle of the debug build binaries
+  target_link_options(cunumeric PRIVATE "${CMAKE_CURRENT_BINARY_DIR}/fatbin.ld")
+endif()
+
+add_custom_target(cunumeric_c_h_c ALL
+  COMMENT "Preprocess src/cunumeric/cunumeric_c.h for Python CFFI bindings"
+  COMMAND ${CMAKE_C_COMPILER} -E -DLEGATE_USE_PYTHON_CFFI
+          "-I$<JOIN:$<TARGET_PROPERTY:legate::core,INTERFACE_INCLUDE_DIRECTORIES>,;-I>"
+          -P "${CMAKE_CURRENT_SOURCE_DIR}/src/cunumeric/cunumeric_c.h"
+          -o "${CMAKE_CURRENT_BINARY_DIR}/include/cunumeric/cunumeric_c.h.i"
+  BYPRODUCTS "${CMAKE_CURRENT_BINARY_DIR}/include/cunumeric/cunumeric_c.h.i"
+  COMMAND_EXPAND_LISTS
+)
+# Generate a target that depends on cunumeric_c_h_c
+add_library(cunumeric_c_h_i INTERFACE)
```

Comment:
Would a more descriptive name like 'cunumeric_c_headers' be better for the library name?